creator-node/src/services/stateMachineManager/index.js

```diff
@@ -133,6 +136,76 @@ class StateMachineManager {
     this.highestEnabledReconfigMode = highestEnabledReconfigMode
     this.enabledReconfigModesSet = enabledReconfigModesSet
   }
+
+    /**
+   * Issues syncRequest for user against secondary, and polls for replication up to primary
+   * If secondary fails to sync within specified timeoutMs, will error
+   */
+     async issueSyncRequestsUntilSynced(
+      secondaryUrl,
+      wallet,
+      primaryClockVal,
+      timeoutMs
+    ) {
+      // Issue syncRequest before polling secondary for replication
+      this.manualSyncQueue.add(
+
+      )
```

Comment:
you can fill this in with something like: 

```js
const { duplicateSyncReq, syncReqToEnqueue } = getNewOrExistingReq(...)
if (!_.isEmpty(duplicateSyncReq)) {
  // TODO: Log duplicate and return or something
} else if (!_.isEmpty(syncReqToEnqueue)) {
  const { jobName, jobData } = syncReqToEnqueue
  await this.manualSyncQueue.add(jobName, jobData)
} else {
  // TODO: Log error that the sync request couldn't be created and return or something
}
```

(import `getNewOrExistingReq` from `stateReconciliationUtils.js`)